packages/@cdktf/hcl2cdk/lib/expressions.ts

```diff
@@ -17,6 +18,317 @@ export type Reference = {
+    }
+    return t.stringLiteral(node.meta.value);
+  }
+
+  if (tfe.isFunctionCallExpression(node)) {
+    const functionName = node.meta.name;
+
+    const argumentExpressions = node.children.map((child) =>
+      convertTFExpressionAstToTs(child, scope, nodeIds, scopedIds)
+    );
+
+    return t.callExpression(t.identifier(functionName), argumentExpressions);
+  }
+
+  if (tfe.isSplatExpression(node)) {
+    const baseExpression = convertTFExpressionAstToTs(
+      node.children[0],
+      scope,
+      nodeIds,
+      scopedIds
+    );
+
+    // We don't convert the relative expression because everything after the splat is going to be
+    // a string
+    let relativeExpression = "";
+    // Check if the splat ended at the .* token
+    if (node.children.length > 1 && node.children[1].meta?.value) {
+      relativeExpression = node.children[1].meta.value;
+    }
+
+    const baseExpressionWithFqn = t.memberExpression(
+      baseExpression,
+      t.identifier("fqn")
+    );
+
+    return t.binaryExpression(
+      "+",
+      baseExpressionWithFqn,
+      t.stringLiteral(node.meta.anonSymbolExpression + relativeExpression)
+    );
+  }
+
+  if (tfe.isConditionalExpression(node)) {
+    const condition = convertTFExpressionAstToTs(
+      node.children[0],
+      scope,
+      nodeIds,
+      scopedIds
+    );
+
+    const trueExpression = convertTFExpressionAstToTs(
+      node.children[1],
+      scope,
+
+      nodeIds,
+      scopedIds
+    );
+
+    const falseExpression = convertTFExpressionAstToTs(
+      node.children[2],
+      scope,
+      nodeIds,
+      scopedIds
+    );
+
+    return t.callExpression(t.identifier("conditional"), [
+      condition,
+      trueExpression,
+      falseExpression,
+    ]);
+  }
+
+  if (tfe.isTupleExpression(node)) {
+    const expressions = node.children.map((child) =>
+      convertTFExpressionAstToTs(child, scope, nodeIds, scopedIds)
+    );
+
+    return t.arrayExpression(expressions);
+  }
+
+  if (tfe.isRelativeTraversalExpression(node)) {
+    const segments = node.meta.traversal.map((tv) => {
+      if (tfe.isNameTraversalPart(tv)) {
+        return tv.segment;
+      }
+      if (tfe.isIndexTraversalPart(tv)) {
+        return tv.key;
+      }
+      return "";
+    });
+
+    // The left hand side / source of a relative traversal is not a proper
+    // object / resource / data thing that is being referenced
+    const source = convertTFExpressionAstToTs(
+      node.children[0],
+      scope,
+      nodeIds,
+      scopedIds
+    );
+
+    // TODO: Replace with lookupNested from https://github.com/hashicorp/terraform-cdk/pull/2672
+    return t.binaryExpression(
+      "+",
+      source,
+      t.stringLiteral("." + segments.join("."))
+    );
+  }
+
+  if (tfe.isForExpression(node)) {
+    const collectionChild = tfe.getChildWithValue(
+      node,
+      node.meta.collectionExpression
+    );
+    if (!collectionChild) {
+      throw new Error("Unable to convert for expression");
```

Comment:
Can you add an example here of a for expression we don't support? And ideally we should create an issue and link it here so folks can leave their use cases here

Reply:
I think this is an unncessary check, since this means that something's gone _really_ wrong with the AST generation for it to not have a child in the AST that's already mentioned in the parent. I've removed it for now, but will think about it more and re-add it if necessary.